Tests/RoslynPluginGeneratorTests/AnalyzerPluginGeneratorTests.cs

```diff
@@ -0,0 +1,361 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="AnalyzerPluginGeneratorTests.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+using System;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarQube.Plugins.Test.Common;
+using System.IO;
+using NuGet;
+using System.Linq;
+using System.Collections.Generic;
+namespace SonarQube.Plugins.Roslyn.RoslynPluginGeneratorTests
+{
+    /// <summary>
+    /// Tests for NuGetPackageHandler.cs
+    /// </summary>
+    [TestClass]
+    public class AnalyzerPluginGeneratorTests
+    {
+        public TestContext TestContext { get; set; }
+
+        private enum License { Required, NotRequired };
+        private enum Node { A, B, C, D, E, F };
+
+        [TestMethod]
+        public void Generate_PackageNoAccept_NoDependencies_Succeeds()
+        {
+            // Arrange
+            TestLogger logger = new TestLogger();
+            string outputDir = TestUtils.CreateTestDirectory(this.TestContext, ".out");
+            
+            string localNuGetSourceDir = TestUtils.CreateTestDirectory(this.TestContext, ".localNuGetSource");
+            string localNuGetDownloadDir = TestUtils.CreateTestDirectory(this.TestContext, ".localNuGetDownload");
+            IPackageManager localNuGetSourceStore = CreatePackageManager(localNuGetSourceDir);
+            SetupTestGraph(localNuGetSourceStore);
```

Comment:
Having had a look at the code, all we really need is a correctly confirgured AnalyzerPluginGenerator, which can be passed as a return from a setup method. Even better, none of the code within the setup method is obviously specific to the testing of requireLiecenseAccept, although later tests might need more than is currently used.